core/controllers/collection_viewer.py

```diff
@@ -33,22 +32,18 @@ class CollectionPage(base.BaseHandler):
     @acl_decorators.can_play_collection
     def get(self, collection_id):
         """Handles GET requests."""
+        collection_rights = rights_manager.get_collection_rights(
```

Comment:
See e.g. [here](https://github.com/oppia/oppia/pull/3591/files#diff-e8e75b25b618133593df5ff9cd67a59bR472). I recommend asking @Arunabh98 about the code if you have questions about it; it is possible to fetch multiple models of different types from the datastore with a single GET operation.

That said, I'll review this PR under the assumption that you'll look into this in a separate pass.